### PR TITLE
Fixed "log_loss_hist" to work with float16

### PR DESCRIPTION
**Pull request by jukofyork**
Fixed `torch.log(1e-10 + losses)` to work with `model_weight_dtype` of `float16`.

`float16` has a minimum sub-normal value of approx `5.96e−8`, so changed to use `1e-8` and now the histogram shows up correctly when using `float16`.

### CODE REVIEW

**Comment by tdrussell** [comment]
Merged. Though do keep in mind, mixed precision and gradient scaling is not implemented. When I train, I always use bfloat16. If you're trying float16, the limited dynamic range may give you problems with training.

**Comment by jukofyork** [comment]
> Merged. Though do keep in mind, mixed precision and gradient scaling is not implemented. When I train, I always use bfloat16. If you're trying float16, the limited dynamic range may give you problems with training.

Yeah, it was a cohere model saved/loaded in `float16` but I was using a `float32` LoRA that showed the problem.

I am now just loading it in `bfloat16` and using a `bfloat16` LoRA with `adamw_kahan`, and it seems to work identically and have around 3-5% speedup from not using `float32`.
